src/m365/spo/commands/sitedesign/sitedesign-get.ts

```diff
@@ -27,24 +31,56 @@ class SpoSiteDesignGetCommand extends SpoCommand {
     return 'Gets information about the specified site design';
   }
 
-  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
-    let spoUrl: string = '';
+  private getSiteDesignId(args: CommandArgs): Promise<string> {
+    if (args.options.id) {
+      return Promise.resolve(args.options.id);
+    }
+
+    const requestOptions: any = {
+      url: `${this.spoUrl}/_api/Microsoft.Sharepoint.Utilities.WebTemplateExtensions.SiteScriptUtility.GetSiteDesigns`,
+      headers: {
+        accept: 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .post<{ value: SiteDesign[] }>(requestOptions)
+      .then(response => {
+        const siteDesignItem: SiteDesign | undefined = response.value.filter(x => x.Title == args.options.title)[0];
+
+        if (!siteDesignItem) {
+          return Promise.reject(`The specified site design does not exist`);
+        }
+
+        if (response.value.filter(x => x.Title == args.options.title).length > 1) {
+          return Promise.reject(`Multiple site designs with title ${args.options.title} found: ${response.value.filter(x => x.Title == args.options.title).map(x => x.Id)}`);
+        }
 
+        return Promise.resolve(siteDesignItem.Id);
+      });
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
     this
       .getSpoUrl(logger, this.debug)
       .then((_spoUrl: string): Promise<ContextInfo> => {
-        spoUrl = _spoUrl;
-        return this.getRequestDigest(spoUrl);
+        this.spoUrl = _spoUrl
+        return this.getRequestDigest(this.spoUrl);
       })
       .then((res: ContextInfo): Promise<string> => {
+        this.formDigestValue = res.FormDigestValue;
+        return this.getSiteDesignId(args);
+      })
+      .then((siteDesignId: string): Promise<string> => {
         const requestOptions: any = {
-          url: `${spoUrl}/_api/Microsoft.Sharepoint.Utilities.WebTemplateExtensions.SiteScriptUtility.GetSiteDesignMetadata`,
+          url: `${this.spoUrl}/_api/Microsoft.Sharepoint.Utilities.WebTemplateExtensions.SiteScriptUtility.GetSiteDesignMetadata`,
           headers: {
-            'X-RequestDigest': res.FormDigestValue,
+            'X-RequestDigest': this.formDigestValue,
```

Comment:
Seems like we don't need the request digest for this request either. I think we can remove this from this command altogether